tests/monitoring_test.go

```diff
@@ -186,6 +229,102 @@ var _ = Describe("[Destructive] Monitoring Tests", func() {
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_datavolumes_pending_for_default_storageclass")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(1))
+
+			By("Check CDINoDefaultStorageClass alert is triggered")
+			waitForPrometheusAlert(f, "CDINoDefaultStorageClass")
+
+			updateDefaultStorageClasses("true")
+
+			By("Ensure metric values after restoring default storage class annotation")
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_storageclasses")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(defaultSCs))
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_virt_storageclasses")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(defaultVSCs))
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_datavolumes_pending_for_default_storageclass")
+			}, metricPollingTimeout, metricPollingInterval).Should(BeZero())
+		})
+
+		It("[test_id:XXXX]CDIDefaultStorageClassDegraded fired when default storage class has no smart clone or ReadWriteMany", func() {
+			if f.SnapshotSCName != defaultStorageClass.Name && f.CsiCloneSCName != defaultStorageClass.Name {
+				Skip("Default storage class does not support snapshot or CSI clone")
+			}
+			if !f.IsSnapshotStorageClassAvailable() && !f.IsCSIVolumeCloneStorageClassAvailable() {
+				Skip("Smart Clone is not applicable")
+			}
+
+			profile := &cdiv1.StorageProfile{}
+			err := f.CrClient.Get(context.TODO(), types.NamespacedName{Name: defaultStorageClass.Name}, profile)
+			Expect(err).ToNot(HaveOccurred())
+
+			hasRWX := false
+			for _, cps := range profile.Status.ClaimPropertySets {
+				for _, am := range cps.AccessModes {
+					if am == corev1.ReadWriteMany {
+						hasRWX = true
+						break
+					}
+				}
+			}
+			if !hasRWX {
+				Skip("Default storage class profile has no ReadWriteMany access mode")
+			}
+
+			By("Ensure initial metric values")
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_storageclass_with_smart_clone_and_rwx")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(1))
+
+			By("Remove storage profile ReadWriteMany access mode")
+			updateDefaultStorageClassProfileClaimPropertySets([]cdiv1.ClaimPropertySet{{
+				AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+				VolumeMode:  &cc.BlockMode,
+			}})
+
+			By("Ensure metric values after updating storage profile")
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_storageclass_with_smart_clone_and_rwx")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(0))
+
+			By("Check CDIDefaultStorageClassDegraded alert is triggered")
+			waitForPrometheusAlert(f, "CDIDefaultStorageClassDegraded")
+
+			By("Restore storage profile")
+			updateDefaultStorageClassProfileClaimPropertySets(nil)
+
+			By("Ensure metric values after restoring storage profile")
```

Comment:
I was about to suggest to ensure that the metric is restored after the cluster state is "fixed"
but I see you already do that in all of the tests. Cool
Testing that alerts go away when things are good is really critical from my experience